src/genomicsdb.pyx

```diff
@@ -48,42 +102,56 @@ def connect(workspace,
     ------
     GenomicsDBException
          On failure to connect to the native GenomicsDB library
-    """    
+    """
     try:
-        return _GenomicsDB(workspace, callset_mapping_file, vid_mapping_file,
-                           attributes, segment_size)
-    except:
-        raise GenomicsDBException("Failed to connect to the native GenomicsDB library")
+        return _GenomicsDB(query_json=query_json, loader_json=loader_json)
+    except Exception as e:
+        print(e)
+        raise GenomicsDBException("Failed to connect to the native GenomicsDB library using json")
 
 cdef class _GenomicsDB:
     cdef GenomicsDB* _genomicsdb
 
-    def __cinit__(self):
-        self._genomicsdb = NULL
-
-    def __init__(self,
-                 workspace,
-                 callset_mapping_file,
-                 vid_mapping_file,
-                 attributes = None,
-                 segment_size = None):
-        cdef string ws = as_string(workspace)
-        cdef vector[string] vec = as_vector(attributes)
-        if attributes is None:
-            self._genomicsdb  = new GenomicsDB(as_string(workspace),
-                                               as_string(callset_mapping_file),
-                                               as_string(vid_mapping_file))
-        elif segment_size is None:
-            self._genomicsdb  = new GenomicsDB(as_string(workspace),
-                                               as_string(callset_mapping_file),
-                                               as_string(vid_mapping_file),
-                                               as_vector(attributes))
+    def __init__(self, **kwargs):
+        if 'query_protobuf' in kwargs  and 'loader_json' in kwargs and kwargs['loader_json'] is not None:
```

Comment:
`'loader_json' in kwargs and kwargs['loader_json'] is not None` could be reduced to `kwargs.get('loader_json', None) is not None`. It saves a few characters, but not a huge deal both seem reasonably readable/Pythonic to my eye.